AI_ML/OPENAI/JSON_EXTRACTOR/JSON_EXTRACTOR.py

```diff
@@ -0,0 +1,83 @@
+from flojoy import flojoy, DataFrame as FlojoyDataFrame, run_in_venv
+import os
+import json
+from copy import deepcopy
+import time
+
+
+ACCEPTED_SCHEMA_FORMATS = [".json"]
+
+BASE_SCHEMA = {
+    "name": "information_extraction",
+    "description": "Extracts the information as JSON.",
+    "parameters": {"type": "object", "properties": {}, "required": []},
+}
+
+API_RETRY_ATTEMPTS = 5
+API_RETRY_INTERVAL_IN_SECONDS = 1
+
+
+@flojoy
+@run_in_venv(pip_dependencies=["openai==0.27.8", "pandas==2.0.2"])
+def JSON_EXTRACTOR(
+    properties: str,
+    prompt: str,
+) -> FlojoyDataFrame:
+    """
+    The JSON_EXTRACTOR node extract specific properties information from a text using JSON schema.
+
+    Parameters
+    ----------
+    properties: string
+        Comma separated list of properties to extract. Example: "name,age,location"
+    prompt: string
+        Text to extract information from. Example: "I'm John, I am 30 years old and I live in New York."
+    """
+    import openai
+    import pandas as pd
+
+    api_key = os.environ.get("OPENAI_API_KEY")
+    if not api_key:
+        raise Exception("OPENAI_API_KEY environment variable not set")
+    openai.api_key = api_key
+
+    if not properties:
+        raise Exception("No properties found to extract.")
+
+    properties = properties.split(",")
```

Comment:
We can enhance the code by utilizing the parameter type `list[str]` or the Flojoy special parameter type `Array`. These options automatically convert comma-separated strings into lists of strings, eliminating the need for reassigning to the same variables and manually splitting the string.